tests/consensus_tests/test_hw_measurement.py

```diff
@@ -163,3 +163,36 @@ def assert_with_upper_bound_error(inp: int, min_value: int, upper_bound_error_pe
 
     if inp > upper_bound:
         assert False, f"Assertion {inp} < {upper_bound} (upperbound) failed. Allowed error = {upper_bound_error_percent}"
+
+
+@pytest.mark.parametrize("test_item", [
+    {"path":"query",
+     "json": {"query": 123, "limit": 10, "with_payload": True,}
+     },
+    {"path":"scroll",
+     "json": {"limit": 10, "with_payload": True,}
+     }], ids=["query", "scroll"])
+def test_payload_io_read_is_within_limit(tmp_path: pathlib.Path, test_item):
+    assert_project_root()
+    # Start cluster
+    env = {
+        "QDRANT__SERVICE__HARDWARE_REPORTING" : "true"
+    }
+    peer_api_uris, peer_dirs, bootstrap_uri = start_cluster(tmp_path, N_PEERS, extra_env=env)
+    # Create collection with 50 segments and 1000 points.
+    # Each point has a payload of roughly 22B.
+    # With a request's limit of 10 points, the payload_io_read should be within 300B
```

Comment:
Maybe let's document what did go wrong before.

```suggestion
    # With a request's limit of 10 points, the payload_io_read should be within 300B
    # Before when reading payloads in every segment it would read 11 kilobytes
    # See: <https://github.com/qdrant/qdrant/pull/6279>
```